### PR TITLE
Updates to Report Seedbanks

### PR DESCRIPTION
**Pull request by charlie-tf**
- Seedbanks have the "selected" field, which we toggle to show whether a
  seedbank is included in the report or not
- Let the seedbank InfoField component use the useStyles hook directly
  instead of passing the className through props
- No longer need two copies, report and draftReport, in ReportEdit
- Do not let the polling for lock status change the report data out from
  under the ReportEdit component

### CODE REVIEW

File: src/components/Reports/ReportEdit.tsx [hunk 1]
```diff
@@ -53,6 +50,16 @@ export default function ReportEdit({ organization }: ReportEditProps): JSX.Eleme
     } else {
       snackbar.toastError(strings.GENERIC_ERROR, strings.REPORT_COULD_NOT_OPEN);
     }
+  }, [reportIdInt, snackbar]);
+
+  const [currentUserEditing, setCurrentUserEditing] = useState(true);
+  useEffect(() => {
```
Comment (charlie-tf) [inline src/components/Reports/ReportEdit.tsx:56]: Moved the `setInterval` into its own `useEffect` and also don't allow its handler to write to the report state. This was changing the state out from under the component and invisibly reverting edits 😱

File: src/components/Reports/SeedbankSection.tsx [hunk 2]
```diff
@@ -148,18 +134,24 @@ export default function SeedbankSection(props: SeedbankSectionProps): JSX.Elemen
...
   value: string | number;
   onChange: (value: any) => void;
-  className: string;
   type: 'text' | 'date';
 };
 
 function InfoField(props: InfoFieldProps): JSX.Element {
-  const { id, label, editable, value, onChange, className, type } = props;
+  const { id, label, editable, value, onChange, type } = props;
+  const classes = useStyles();
```
Comment (charlie-tf) [inline src/components/Reports/SeedbankSection.tsx:154]: This class didn't have to be passed in through props. Use `useStyles` directly in this component.

File: src/components/Reports/ReportEdit.tsx [hunk 3]
```diff
@@ -226,15 +236,14 @@ export default function ReportEdit({ organization }: ReportEditProps): JSX.Eleme
...
             (showAnnual ? (
               <ReportFormAnnual editable={true} report={report} />
             ) : (
               <ReportForm
                 editable={true}
-                draftReport={draftReport}
+                draftReport={report}
                 onUpdateReport={updateReport}
-                allSeedbanks={getAllSeedBanks(organization).map((f) => ReportService.seedbankFromFacility(f, report))}
+                allSeedbanks={report.seedBanks ?? []}
```
Comment (charlie-tf) [inline src/components/Reports/ReportEdit.tsx:246]: `report.seedBanks` will always contain the full list of seedbanks.

File: src/components/Reports/SeedbankSection.tsx [hunk 4]
```diff
@@ -169,6 +161,11 @@ function InfoField(props: InfoFieldProps): JSX.Element {
       <></>
     )
   ) : (
-    <OverviewItemCard isEditable={false} title={label} contents={value.toString() ?? '0'} className={className} />
+    <OverviewItemCard
+      isEditable={false}
+      title={label}
+      contents={value.toString() ?? '0'}
+      className={classes.infoCardStyle}
```
Comment (karthikbtf) [inline src/components/Reports/SeedbankSection.tsx:168]: cool

File: src/components/Reports/ReportEdit.tsx (hunk 1 above)
Comment (karthikbtf) [reply src/components/Reports/ReportEdit.tsx:56]: good catch!

**Review by karthikbtf — APPROVED** [review]
👍

**Pull request merged**